assets/scripts/segments/EmptySegment.jsx

```diff
@@ -5,76 +5,50 @@ import MeasurementText from '../ui/MeasurementText'
 import { TILE_SIZE } from '../segments/view'
 import { t } from '../app/locale'
 
-class EmptySegment extends React.Component {
+class EmptySegment extends React.PureComponent {
   static propTypes = {
     remainingWidth: PropTypes.number,
     occupiedWidth: PropTypes.number,
     units: PropTypes.number,
     position: PropTypes.string
   }
 
-  constructor (props) {
-    super(props)
-
-    this.state = {
-      widthValue: 0,
-      segmentWidth: 0
-    }
-  }
-
-  componentDidUpdate (prevProps) {
-    const { remainingWidth } = this.props
-    if (remainingWidth && prevProps.remainingWidth !== remainingWidth) {
-      this.repositionEmptySegment()
-    }
-  }
-
-  hideEmptySegment = () => {
-    this.streetEmptySegment.classList.remove('visible')
-  }
-
-  showEmptySegment = (width) => {
-    this.streetEmptySegment.classList.add('visible')
-
-    this.setState({
-      widthValue: width / TILE_SIZE,
-      segmentWidth: (this.props.position === 'right') ? width - 1 : width
-    })
-  }
-
   repositionEmptySegment = () => {
```

Comment:
Since this function now takes initial values and returns a value, we're on the way to making this more of a functional method, so we might as well go all the way. (Previously, this couldn't be functional because it returned nothing and resolved with side effects (setting state)). (We also probably did not do this consistently everywhere, but I'm starting to learn how to identify these opportunities, besides, they make components easier to test*)

Instead of reading `this.props`, we could pass in the same values as arguments:

```js
repositionEmptySegment (remainingWidth, occupiedWidth, position) {
```

Notice this means we don't need to bind `this` to the method anymore with an arrow function.

> *they make components easier to test

The first thing is that the method is just easier to unit-test. Given certain values, make sure our render is going to look a certain way (especially in the special case of what happens when the entire street is empty).

The other thing I learned is that you can't ask Enzyme/Jest to [spy on an arrow function in a class](https://github.com/airbnb/enzyme/issues/365)! While we probably don't need to spy on this method here, I'm thinking of using these arrow functions only when we need it, and not as a knee jerk default.